services/harmony/app/backends/workflow-orchestration/work-item-updates.ts

```diff
@@ -698,6 +696,14 @@ export async function processWorkItem(
       return;
     }
 
+    const hasOutput = results?.length > 0;
```

Comment:
I'm surprised this line doesn't give you a compiler error. if results can be null then `results?.length` will be null, which can't be used with `>`

Reply:
I think
```
const hasOutput = results &&  results.length > 0
```
is better

Reply:
The ? is shorthand for what you wrote.

Reply:
I was wrong, James is correct - well mostly - the documentation says it will short circuit to undefined rather than null: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining